lib/rules/asi-safe.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview Rule to flag lines dangerous when relying on ASI
+ * @author Glen Mailer
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+module.exports = function(context) {
+
+    function report(node, token) {
+        context.report(node, {
+            line: token.loc.start.line,
+            column: token.loc.start.column
+        }, "Line starting with " + token.value + " continues previous");
+    }
+
+    function check(node) {
+        var tokens = context.getTokensBefore(node, 2);
+        var paren = tokens[1];
+        var before = tokens[0];
+        if (paren.loc.start.line !== before.loc.end.line) {
+            report(node, paren);
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Public API
+    //--------------------------------------------------------------------------
+
+    return {
+
+        "MemberExpression": function(node) {
+            if (!node.computed) {
+                return;
+            }
+
+            check(node.property);
+        },
+
+        "CallExpression": function(node) {
+            if (!node.arguments[0]) {
```

Comment:
This is a strange case. Are you just testing to see if there are no parameters? If so, use `length`